src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java

```diff
@@ -208,6 +221,17 @@ private void validateDateRange(FailureCollector collector) {
     }
   }
 
+  private void validatePageSize(FailureCollector collector) {
+    if (containsMacro(ServiceNowConstants.PROPERTY_PAGE_SIZE)) {
+      return;
+    }
+
+    if (getPageSize() < 0 || getPageSize() > 10000) {
```

Comment:
should be `if (getPageSize() <= 0 || getPageSize() > 10000) {`? `int pages = (tableInfo.getRecordCount() / pageSize);` - this line divide by 0.

Reply:
Good catch! 
Refactored it. 84fa2da26d0a11988d339f44dd6c112e87cff819